.github/actions/yarn-install/action.yml

```diff
@@ -0,0 +1,23 @@
+name: Install Dependencies
+inputs:
+  node-version-type:
+    required: false
+    default: 'lts'
+    type: choice
+    options:
+      - 'lts'
+      - 'min-supported'
+
+runs:
+  using: "composite"
+  steps:
+  - uses: actions/setup-node@v4
+    with:
+      # if both node-version-file and node-version are provided, node-version is used
+      node-version-file: 'package.json'
+      node-version: ${{ inputs.node-version-type == 'lts' && 'lts/*' || ''  }}
```

Comment:
I don't fully understand the use of `node-version-file`/`node-version` here - can we not use a matrix like the example at https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/using-a-matrix-for-your-jobs#example-using-a-single-dimension-matrix ?

Reply:
I've updated the comment. please see if you agree with this approach. I can certainly hardcode the versions, but then the minimum supported version on `package.json` will be maintained separately from that hardcoded version in the pipelines, which is not the end of the world, but with no single point of truth they might diverge.

Reply:
My main concern here is the CI configuration effectively leaking into the repo (the `volta` field) in a way that's not obvious if you're looking at it from the `package.json` side. And, if you're a contributor actually using Volta, this is effectively breaking by forcing you to develop using 18.0.0, right?

> with no single point of truth they might diverge.

This is a fair point but I don't see it as a disaster:

 - Bumping the CI minimum without bumping Metro's minimum isn't a concern, I can't see a reason we'd ever do that voluntarily.
 - If we bumped Metro's minimum but forgot the CI minimum, the worst that happens is CI breaks some time later when we actually break compatibility, so we notice and update it. 
 - We probably *should* have an E2E test that builds and publishes, then installs Metro using something like Verdaccio, like RN does - then we'd fail fast because the `engines` requirement would cause installation to fail.
 
There are upsides to hardcoding the matrix - it's cleaner and more GHA-native (automatically reflected in test names, etc). I think that'd be my preference tbh. `['current', 'lts/*', 'lts/-1', '18.0']` would give us good coverage.

Reply:
gotcha. I also spoke with @huntie about that.
could you please take a look at the updated code?